src/MessageHandler/DatasetSubmissionFilerHandler.php

```diff
@@ -142,16 +143,17 @@ public function __invoke(DatasetSubmissionFiler $datasetSubmissionFiler)
                 $filesInfo[$file->getId()]['filePathName'] = $file->getFilePathName();
                 $filesInfo[$file->getId()]['physicalFilePath'] = $file->getPhysicalFilePath();
             }
-            $this->logger->info('Zipfile opened: ' . $destinationPath);
+            $this->logger->info('Zipfile opened: ' . $destinationPath, array_merge($loggingContext, array('PHP_memory_usage' => memory_get_usage())));
```

Comment:
Maybe you want to do this outside this function? Now it will only be used in this particular log. Which may be intentional?

Reply:
The intent was to capture any change in memory consumption, so the memory_get_usage has to be called right before and after each file.  I also called after zipfile->finish to ensure any allocated memory is released, or to otherwise try and detect memory leaks. I figured in the capture, the better, until we get to the bottom of what's happening.